### PR TITLE
Remove word recovery

### PR DESCRIPTION
**Pull request by keepkeyjon**

### CODE REVIEW

File: keepkeyctl [hunk 1]
```diff
@@ -341,7 +341,7 @@ class Commands(object):
         return self.client.wipe_device()
 
     def recovery_device(self, args):
-        return self.client.recovery_device(args.use_trezor_method, args.words, args.passphrase_protection,
+        return self.client.recovery_device(False, args.words, args.passphrase_protection,
```
Comment (keepkeyjon) [inline keepkeyctl]: I left the signature of this alone in case Electrum are using it.

**Comment by 3rdIteration** [comment]
Is there a particular reason why this was removed?

**Comment by keepkeyjon** [comment]
It was removed from the firmware due to space constraints https://github.com/keepkey/keepkey-firmware/issues/190

**Comment by 3rdIteration** [comment]
And the new method of doing a secure restore or verifying recovery phrase will be?

**Comment by keepkeyjon** [comment]
Ciphered recovery isn't new.

**Comment by 3rdIteration** [comment]
I'll rephrase. What is replacement method that allows a user securely recover an existing seed or verify an existing one in a non-destructive way if the ciphering is removed as yesterday's release?

**Comment by 3rdIteration** [comment]
> It was removed from the firmware due to space constraints [keepkey/keepkey-firmware#190](https://github.com/keepkey/keepkey-firmware/issues/190)

Ok, this will teach me to look at this stuff on my phone, read it that both were being removed... (Which seemed inconceivable, good thing I was mistaken, apologies for the mix-up) The confusion here is that the Trezor style word recovery has been removed, but the cipher recovery is still there, but has been broken in terms of how it is accessed via both Electrum and KeepkeyCTL in the new keepkeylib.

The issue is that the logic in client.py is backwards in terms of how it treats use_trezor_method, allowing it to run using the old style if use_trezor_method == true but blocks it if use_trezor_method == false.

Created a PR https://github.com/keepkey/python-keepkey/pull/85

**Comment by keepkeyjon** [comment]
Ah, I see... oops. Thanks for the PR.

**Comment by SomberNight** [comment]
Nit: also re `word_count`, I guess it is now ignored by the firmware then? And the check/raise exception branch no longer makes sense? Seems mostly harmless though -- except if someone was e.g. explicitly passing 0 when using the cipher recovery method will now get an exception.

-----

old code:
https://github.com/keepkey/python-keepkey/blob/aadf95737d6554610abf8a7b2d0e0f1a1b37f80e/keepkeylib/client.py#L1064-L1067
new code:
https://github.com/keepkey/python-keepkey/blob/2170fc4479bf5a4b63d2067801ecf041430db769/keepkeylib/client.py#L1118-L1121

**Comment by keepkeyjon** [comment]
Firmware now supports two modes when doing ciphered recovery:

1) word_count == 0, or not provided
  Variable length seed phrase, length enforced to 12/18/24 words once the user has entered all their words. KeepKey client does this, and it's pretty bad UX.
2) word_count in (12, 18, 24)
  Fixed length, decided up-front.

But it looks as if I missed exposing that on the python-keepkey side of things. If Electrum prefers (1), we can fix that & spin another release.

**Comment by SomberNight** [comment]
No need to do a release just for this.

**Pull request merged**